ladybug_geometry/geometry2d/polygon.py

```diff
@@ -843,3 +899,96 @@ def __eq__(self, other):
 
     def __repr__(self):
         return 'Polygon2D ({} vertices)'.format(len(self))
+
+    def perimeter_zones(self, distance, tol=1e-10):
```

Comment:
@saeranv ,
Can we pull this and the other polyskel methods out into their own module in `ladybug_geometry_polyskel`? Then they would just accept accept a `Polygon2D` as input and we can keep the dependency unidirectional. You can see that this is also why the tests are failing.

Reply:
Also, I know that our first main application og this code is likely to be generating zone geometry but, while we are in a purely geometry library, can we use a more generic/abstract term in case we wind up using this method for other applications?  Is there some term from the polyskeleton literature that we could use instead of "zone"? Like "cells" or "sub_polygons"?

Reply:
@chriswmackey Makes sense, I'll move it to the polyskel repo. As for naming, Felkel and Obdrzalek refer to them as sub-polygons. It tends not to come up much since straight skeletons tend to be discussed purely in terms of it's edge networks (which is why the original algorithm just generates a set of unordered edges).

I can change these methods to perimeter_sub_polygons and core_sub_polygons.